### ISSUE TITLE
Feature request: Do not reduce row height when zooming out

### ISSUE THREAD

**Issue by edgardomortiz** [AliView/AliView#72]
It would be useful to be able to keep the sequence height stable while zooming out, especially when the alignment has few sequences (something like Geneious alignment viewer).

**Comment by willfischer** [comment]
Yes, yes, yes!  Specifically, locking the height of the alignment display to the window height would be extremely helpful.

— Will Fischer

______________________
Will Fischer, PhD

net: wfischer@lanl.gov
tel: 505-665-4149
fax: 505-665-3493

Group T-6, Theoretical Biology
MS K710, Drop Point 03041003U
Los Alamos National Laboratory
Los Alamos, NM 87545
USA

**Comment by ewallace** [comment]
I'm using aliview a lot this month (thank you!) and this is the feature that I want the most. Independent zoom controls for the width/position scaling from the height/sequence scaling would be amazing. It enables:
* easier visual comparison of alignments with different numbers of sequences
* readable sequence names while viewing alignments of wide features.

This is especially a problem in the absence of vector image exports (#22). If aliview did native vector image exports, then we could resize the output in a vector graphics program afterwards without loss of quality. Currently, changing the aspect ratio of the .png export results in resampling and a blurred image.

Edward

**Comment by willfischer** [comment]
This is a critical lack at present. When working with "wide" alignments (say, coronavirus genomes) where there are many more columns than rows, it would be fabulously useful to be able to (in effect) squeeze the view from the sides, while not losing height. As it is, when you have, say 5 sequences, by the time you have 5,000 columns on the screen, you can't see anything because the vertical scale has shrunk away.


Fixing this would mean allowing the aspect ratio of the per-base display rectangle to change, or possibly allowing each base to have multiple rectangles (stacked one over the other).  Without knowing the details of the display model, it's hard to suggest strategies, but this is a really important feature.


Thanks for the program, which is excellent in most respects and essential to our work.


-- Will


______________________
Will Fischer, Ph.D.

net: wfischer@lanl.gov
tel: 505-665-4149
fax: 505-665-3493

Group T-6, MS K710, Drop Point 03041003U
Los Alamos National Laboratory
Los Alamos, NM 87545
USA


________________________________

**Comment by andersla** [comment]
I hope that I will have time to work on AliView again Jun-Aug, I guess this is issue one of the top priorities.

**Comment by ewallace** [comment]
@andersla I could talk to our university's Research Software Engineer group and see if there's anyone with the expertise to add and pull request these features. I have some discretionary money on a grant that could be used for this sort of thing. What do you think?

**Comment by willfischer** [comment]
Another option for working on "wide" alignments (where the number of columns >> the number of rows) would be to wrap the view on the screen ...



______________________
Will Fischer, Ph.D.

net: wfischer@lanl.gov
tel: 505-665-4149
fax: 505-665-3493

Group T-6, MS K710, Drop Point 03041003U
Los Alamos National Laboratory
Los Alamos, NM 87545
USA


________________________________

**Comment by andersla** [comment]
@ewallace Pull requests are very welcome! But I have to warn that this very feature is not all simple I think, it is affecting the basics of the whole painting and scaling of alignments.
